File: kotlin-sdk/src/main/kotlin/com/cleanster/model/ApiResponse.kt
package com.cleanster.model

/**
 * Standard response envelope returned by every Cleanster API endpoint.
 *
 * @param T The type of the `data` payload.
 * @property status HTTP-style status code (200, 400, 401, 404, 500).
 * @property message Human-readable status description.
 * @property data The response payload; `null` on errors.
 */
data class ApiResponse<T>(
    val status:  Int,
    val message: String,
    val data:    T?
)

File: swift-sdk/Sources/Cleanster/Models/ApiResponse.swift
import Foundation

/// The standard response envelope returned by every Cleanster API endpoint.
public struct ApiResponse<T: Decodable>: Decodable {
    /// HTTP-style status code (200, 400, 401, 404, 500).
    public let status: Int
    /// Human-readable status description.
    public let message: String
    /// The response payload. `nil` on errors.
    public let data: T?
}
